api_core/google/api_core/grpc_helpers.py

```diff
@@ -157,35 +160,38 @@ def __init__(self, method, channel):
         exception, the exception will be raised when this is invoked.
         """
         self.responses = None
-        """Iterator[protobuf.Message]: A iterator of responses. If specified,
-        self.response will be populated on each invocation by calling
-        ``next(self.responses)``."""
+        """Iterator[
+            Union[protobuf.Message, Callable[protobuf.Message], exception]]:
+        An iterator of responses. If specified, self.response will be populated
+        on each invocation by calling ``next(self.responses)``."""
         self.requests = []
         """List[protobuf.Message]: All requests sent to this callable."""
         self.calls = []
         """List[Tuple]: All invocations of this callable. Each tuple is the
         request, timeout, metadata, and credentials."""
 
     def __call__(self, request, timeout=None, metadata=None, credentials=None):
-        self._channel._requests.append((self._method, request))
-        self.requests.append(request)
+        self._channel.requests.append(
+            _ChannelRequest(self._method, request))
         self.calls.append(
-            _ChannelCall(request, timeout, metadata, credentials))
+            _MethodCall(request, timeout, metadata, credentials))
+        self.requests.append(request)
 
-        if self.responses:
-            self.response = next(self.responses)
+        response = self.response
+        if response is None and self.responses is not None:
```

Comment:
If you **want** you could do validation here:

```python
if self.responses is not None:
    if response is None:
        response = next(self.responses)
    else:
        raise ValueError('Why you collision')
```

Reply:
Donion rings.